executor/explain.go

```diff
@@ -73,14 +73,14 @@ func (e *ExplainExec) Next(ctx context.Context, req *chunk.Chunk) error {
 }
 
 func (e *ExplainExec) generateExplainInfo(ctx context.Context) (rows [][]string, err error) {
-	closed := false
-	defer func() {
-		if !closed && e.analyzeExec != nil {
-			err = e.analyzeExec.Close()
-			closed = true
-		}
-	}()
 	if e.analyzeExec != nil && !e.executed {
+		closed := false
+		defer func() {
+			if !closed && e.analyzeExec != nil {
+				err = e.analyzeExec.Close()
+				closed = true
+			}
+		}()
```

Comment:
This `defer` is duplicate to line 93. I think they can be merged into `defer` part and the `closed` variable can also be eliminated.

Reply:
done.

Reply:
done.

Reply:
No, it will fail in test `TestApplyCache` and `TestExplainAnalyze`.